components/tidb_query_expr/src/impl_time.rs

```diff
@@ -1606,6 +1634,38 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_add_date_and_string() {
+        let mut ctx = EvalContext::default();
+        let cases = vec![
+            (Some("2021-03-26"), Some("00:00:00"), Some("2021-03-26")),
+            (Some("2021-03-26"), Some("23:59:59"), Some("2021-03-26")),
+            (Some("2021-03-26"), Some("24:00:01"), Some("2021-03-27")),
+            (Some("2021-03-26"), Some("48:00:01"), Some("2021-03-28")),
```

Comment:
Please add case like:

```
                Some("2018-12-31 23:00:00"),
                Some("1 01:30:30"),
```

and

```
                Some("2019-01-01 01:00:00"),
                Some("-01:01:00"),
```

Reply:
Maybe we can assume the first param does not contain time?
Anyway, there should be a case whose first param is a valid date and the second param is an invalid time.

Reply:
done ptal